operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java

```diff
@@ -30,16 +30,21 @@ public abstract class AbstractDependentResource<R, P extends HasMetadata>
   private ResourceDiscriminator<R, P> resourceDiscriminator;
   private final DependentResourceReconciler<R, P> dependentResourceReconciler;
 
-  protected String name = DependentResource.defaultNameFor(this.getClass());
+  protected String name;
 
   @SuppressWarnings({"unchecked"})
   protected AbstractDependentResource() {
+    this(null);
```

Comment:
`DependentResource.defaultNameFor(this.getClass())`  could be passed here

Reply:
No, this actually doesn't work (I tried, iirc) because you cannot call `this.getClass()` until the chained constructor is called. This will be possible in Java 22, though, I think.

Reply:
Ahh ok , wishful thiking... :) thx